src/components/structures/RoomView.js

```diff
@@ -572,6 +588,15 @@ module.exports = React.createClass({
         });
     },
 
+    _onSearchResultSelected: function(searchResult) {
+        this.setState({searchContext: searchResult.context});
+    },
+
+    getSearchContextTiles: function() {
+        return this._timelineToTiles(
+            this.state.searchContext.getTimeline(),
+            !!this.state.searchContext.startToken);
```

Comment:
Done 25e9376. Also s/`startToken`/`getPaginateToken(true)`/, to make this actually work.